### ISSUE TITLE
InputMedia doesn't apply caption_entities

### ISSUE THREAD

**Issue by dsaltyfrere** [aiogram/aiogram#575]
## Context

edit_message_media does not correct/edit/maintain formatting that's created using telegram's built-in formatting functionality (e.g. ctrl+k for a hyperlink). This may specifically be towards hyperlinks as i have not tested out other shortcuts.

* Operating System: Debian 10 / Ubuntu 20.04
* Python Version: 3.8.2
* aiogram version: 2.13
* aiohttp version: 3.7.4
* uvloop version (if installed): 0.14.0  

## Expected Behavior

I have a bot which allows a user to send a message in a channel and foward it to other channels subscribed on the bot. When the user uses telegram's built-in formatting functionality, e.g. use ctrl+k to create a hyperlink, when edit_media_message is called the formatting is removed despite the message clearly containing the caption_entities object.

## Current Behavior

Formatting is removed/purged once the user edits the post and the post is corrected in subscribed channels ( the formatting remains in the operator channel )

## Failure Information (for bugs)

Please help provide information about the failure if this is a bug. If it is not a bug, please remove the rest of this template.

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Send a photo using caption_entities, entities built-in from telegram ( no HTML/Markdown), use ctrl+k to create a hyperlink.
2. Edit the photo, add some text to the message caption
3. Observe result in forwarded message, the existing caption entity from step 1) will be removed/purged.

### Failure Logs
INFO:__main__:{"message_id":20,"sender_chat":{"id":-1001456841024,"title":"MyTestChannel","type":"channel"},"chat":{"id":-1001456841024,"title":"MyTestChannel","type":"channel"},"date":1619867741,"edit_date":1619867766,"photo":[{"file_id":"AgACAgQAAx0CVtWhQAADFGCNOHYR5nHMYzlpjhUjRy3pmlKcAAKSuDEbsxRpUGLXOCmgKsb7pIJELl0AAwEAAwIAA20AA_zOAAIfBA","file_unique_id":"AQADpIJELl0AA_zOAAI","file_size":6921,"width":320,"height":89},{"file_id":"AgACAgQAAx0CVtWhQAADFGCNOHYR5nHMYzlpjhUjRy3pmlKcAAKSuDEbsxRpUGLXOCmgKsb7pIJELl0AAwEAAwIAA3gAA_3OAAIfBA","file_unique_id":"AQADpIJELl0AA_3OAAI","file_size":18625,"width":730,"height":202}],"caption":"Img caption URL t","caption_entities":[{"type":"text_link","offset":12,"length":3,"url":"https:\/\/www.google.com\/"}]}
{"type":"text_link","offset":12,"length":3,"url":"https:\/\/www.google.com\/"}

*_Please let me know if any additional information is required or if i have omitted important information that is necessary._*

**Comment by Olegt0rr** [comment]
@dsaltyfrere, please provide your code with a second step

**Comment by dsaltyfrere** [comment]
Hi @Olegt0rr ,

Please find the code sample below. This method is called when a message is editted in the main channel and the edits happen in other channels which are subscribed.

Using logger/print statements shows `entities= message['caption_entities'][-1]` is not empty (see logs in OP) but once the edits are done the formatting dissapears.
Oddly enough this is only the case for `ContentType.PHOTO`, in the case of `ContentType.TEXT` this works fine.

```python
async def edit_forward(message: types.Message, forward: Forward) -> types.Message:
    type = message.content_type
    bot = message.bot
    if type is ContentType.TEXT:
        return await bot.edit_message_text(
            message.html_text,
            forward.group_id,
            forward.telegram_id,
            disable_web_page_preview=disable_preview(message),
        )
    elif type is ContentType.PHOTO:
        photo = message.photo[-1]
        entities = message['caption_entities'][-1]
        media = InputMediaPhoto(photo.file_id, caption=message.caption, caption_entities=entities) 
        return await bot.edit_message_media(
            media, forward.group_id, forward.telegram_id
        )
    else:
        logger.error('Could not determine ContentType for %s', message)
```

If any additional information is required please let me know.

Kind regards

**Comment by dsaltyfrere** [comment]
Hi @Olegt0rr,

Using the example provided by you: 
```python
....
entities = message.caption_entities
media = InputMediaPhoto(photo.file_id, caption=message.caption, caption_entities=entities)
return await bot.edit_message_media(media, forward.group_id, forward.telegram_id)
```
Threw the following error:
<aiogram.types.message_entity.MessageEntity object at 0x7f5ea184c460> is not JSON serializable.

Fortunately using `InputMediaPhoto(photo.file_id, caption=message.html_text` achieves the expected result.

Thanks for the assistance.

**Comment by MasterGroosha** [comment]
Please re-open. This fix breaks album sending: https://pastebin.com/nNACg29r

See https://t.me/aiogram_ru/680074 (Russian)

**Comment by Olegt0rr** [comment]
@MasterGroosha traceback?

**Comment by MasterGroosha** [comment]
@Olegt0rr with aiogram v2.14.1

Test code:
```python
import asyncio
from aiogram import Bot, Dispatcher, types


CHAT_ID = 1234567890
PATH_TO_IMAGE = "testfile.jpg"
BOT_TOKEN = "322223322:abcdefghijklmnop"


bot = Bot(token=BOT_TOKEN)
dp = Dispatcher(bot)


async def main():
    media = types.MediaGroup()
    media.attach_photo(types.InputMediaPhoto(PATH_TO_IMAGE))
    await bot.send_media_group(chat_id=CHAT_ID, media=media)
    await bot.session.close()

asyncio.run(main())
```

```
Traceback (most recent call last):
  File "xxx/aiogram-anything.py", line 21, in <module>
    asyncio.run(main())
  File "/usr/local/lib/python3.8/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "xxx/aiogram-anything.py", line 16, in main
    media.attach_photo(types.InputMediaPhoto(PATH_TO_IMAGE))
  File "xxx/venv/lib/python3.8/site-packages/aiogram/types/input_media.py", line 200, in __init__
    super().__init__(
  File "xxx/venv/lib/python3.8/site-packages/aiogram/types/input_media.py", line 49, in __init__
    super(InputMedia, self).__init__(*args, **kwargs)
  File "xxx/venv/lib/python3.8/site-packages/aiogram/types/base.py", line 95, in __init__
    self.props[key].set_value(self, value, parent=self)
  File "xxx/venv/lib/python3.8/site-packages/aiogram/types/fields.py", line 57, in set_value
    value = self.deserialize(value, parent)
  File "xxx/venv/lib/python3.8/site-packages/aiogram/types/fields.py", line 141, in deserialize
    for item in value:
TypeError: 'NoneType' object is not iterable
```

If you manually revert #583 (comment out `caption_entities: typing.List[MessageEntity] = fields.ListField(base=MessageEntity)`), this is what you get instead:

```
Traceback (most recent call last):
  File "xxx/aiogram-anything.py", line 21, in <module>
    asyncio.run(main())
  File "/usr/local/lib/python3.8/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "xxx/aiogram-anything.py", line 17, in main
    await bot.send_media_group(chat_id=CHAT_ID, media=media)
  File "xxx/venv/lib/python3.8/site-packages/aiogram/bot/bot.py", line 1017, in send_media_group
    result = await self.request(api.Methods.SEND_MEDIA_GROUP, payload, files)
  File "xxx/venv/lib/python3.8/site-packages/aiogram/bot/base.py", line 208, in request
    return await api.make_request(self.session, self.server, self.__token, method, data, files,
  File "xxx/venv/lib/python3.8/site-packages/aiogram/bot/api.py", line 140, in make_request
    return check_result(method, response.content_type, response.status, await response.text())
  File "xxx/venv/lib/python3.8/site-packages/aiogram/bot/api.py", line 115, in check_result
    exceptions.BadRequest.detect(description)
  File "xxx/venv/lib/python3.8/site-packages/aiogram/utils/exceptions.py", line 140, in detect
    raise err(cls.text or description)
aiogram.utils.exceptions.InvalidHTTPUrlContent: Failed to get http url content
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x7f0d5089fb80>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x7f0d508d4a60>, 3572.998037027)]']
connector: <aiohttp.connector.TCPConnector object at 0x7f0d5089fa90>

Process finished with exit code 1
```

Finally, if you replace `types.InputMediaPhoto` with `types.InputFile`, everything works fine.

**Comment by Olegt0rr** [comment]
Fixed by #622
